internal/encryption/base.go

```diff
@@ -117,7 +117,7 @@ func (s *baseEncryption) encrypt(data []byte) ([]byte, error) {
 		Meta:    s.encMeta,
 		Data:    encd,
 	}
-	jsond, err := json.Marshal(es)
+	jsond, err := json.Marshal(enhance(es))
```

Comment:
It feels kinda hacky. If there are more use-cases, we should consider delegating serialization and validation outside of `baseEncryption` since they are already "leaked"

Reply:
I'm hoping that long-term we can revert this PR, but I'm not sure that it's realistic to do so as there are many many http backends.

Reply:
The alternate (without a larger refactor) would be doing a unmarshal/enchance/marshal inside of state.go, which adds a lot of overhead.

Reply:
I agree that the current implementation is the most optimal. Let's see where it goes in the future, thanks!